subsys/spm/secure_services.c

```diff
@@ -47,6 +50,34 @@ int spm_secure_services_init(void)
 }
 
 
+#ifdef CONFIG_SPM_SERVICE_READ
+__TZ_NONSECURE_ENTRY_FUNC
+int spm_request_read(void *destination, const void *source, size_t num)
+{
+	static const u32_t read_ranges[] = {
+#ifdef PM_MCUBOOT_ADDRESS
+		/* Allow reads of mcuboot metadata */
+		PM_MCUBOOT_PAD_ADDRESS,
+		PM_MCUBOOT_PAD_ADDRESS + PM_MCUBOOT_PAD_SIZE
+#endif
+	};
+
+	if (destination == NULL || num <= 0) {
+		return -EINVAL;
+	}
+
+	for (int i = 0; i < sizeof(read_ranges); i += 2) {
```

Comment:
read_ranges will be structural this code will looks much readable (see my comment on read_range declaration)